README.md

```diff
@@ -1,3 +1,388 @@
+and `pie install xdebug` would be equivalent.
+
+If `version-constraint` is given, try to install that version if it matches the allowed versions.
+
+On Windows this downloads the correct DLL (attached as file to the release tag), if available. Otherwise, it will
+download the source and compile the extension.
+
+It will then attempt to create a 20-{extension_name}.ini file with `extension={extension_name}` in the directory
+returned by `php-config –ini-dir`, the name that is used is the one from the "name" element in the metadata file.
+
+If `@dev-branch-name` is given, try to install from the branch called "branch-name", for example use `@dev-master` to
+compile from your master branch. As branches would usually not have binaries, this would not work for Windows.
+
+You can't use both a `version-constraint` and `@dev-branch-name` at the same time.
+
+Each step should run with as few privileges as possible.
+
+#### `build {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Same behaviour as install, but only compiles (or downloads if it's Windows).
+
+#### `changelog {?version}`
+
+Shows the release notes of the version it was going to install, or from the specific version, if given.
+
+#### `download {ext-name}{?:version-constraint}{?@dev-branch-name}`
+
+Same behaviour as build, but puts the files in a local directory for manual building and installation.
+
+#### `info`
+
+Shows the description from the metadata file
+
+#### `list`
+
+Shows all available commands
+
+#### `show`
+
+Shows all installed extensions available with the PHP version in the path, including their versions. This includes
+*all* loaded PHP extensions, and not just PIE-sourced ones.
+
+#### `upgrade {?{ext-name}{?:version-constraint}{?@dev-branch-name}}`
+
+Attempts to upgrade all installed versions to the latest available ones on GitHub (unless their major version has
+changed) If a specific package is specified, only attempt to upgrade that specific one.
+
+### CLI Options
+
+#### `--dry-run`
+
+Shows what it is about to do, but doesn't actually install
+
+#### `--force`
+
+To attempt to install a version that doesn't match the version constraints from the meta-data, for instance to install
+an older version than recommended, or when the signature is not available.
+
+#### `--with-php-config=/path/to/php-config`
+
+Allows installation of extensions with PHP versions that are not in the path
+
+#### `--{option}{?=value}`
+
+All options specified in the `config` section of the composer.json file can also be given, including a value if they
+take them. For example, for Xdebug you could run `pie install xdebug --without-xdebug-compression`.
+
+## Extension maintainer: register a PIE package
+
+Create a `composer.json` in your repository, commit:
+
+```json
+{
+    "name": "ext-xdebug",
+    "type": "php-ext-zend",
+    "license": "Xdebug-1.03",
+    "description": "Xdebug is an amazing tool...",
+    "require": {
+        "ext-zlib": "*",
```

Comment:
@asgrim 

Should this not be zlib instead of ext-zlib as it is a library? For extensions as dependency we can use the ext prefix, for example in phalcon, we will have ext-psr as dependency.

Reply:
Yep as @derickr mentioned, is really just an example of depending on another extension (not a library). I will make it clearer that we are not having lib dependencies

Reply:
Updated